### ISSUE TITLE
http redirect's initial connection won't connect

### ISSUE THREAD

**Issue by oliv3r** [yrutschle/sslh#385]
When start sslh, and having a configuration such as:
```
listen: ( { host: "0.0.0.0"; port: "8080"; } )
protocols: { name: "http"; host: "<ip>"; port: "80"; }, )
```

The first connection just hangs (e.g. `wget --spider -q http://localhost:8080`). While all is running, coming from an outside host (somewhere around the world), it gets an immediate reply, but also 'unhangs' the first one.

Doing it the other way around, connecting from around the world, the connection hangs. Connecting internally, also hangs, but it unblocks the external connection which now gets a reply. The internal one this time, sits there trying, until the external is up again, which then unblocks the internal one. This both 8080, but also 443.

So in short;
localhost -> external: hang, but release
external -> localhost: hang, hang, release

btw, the external host can connect over any (configured) port to unhang the internal connection.
